torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -2270,6 +2270,66 @@ Tensor eig_backward(const std::vector<torch::autograd::Variable> &grads, const T
   return at::linalg_solve(Uh, at::matmul(U_contrib, Uh) + D_contrib * Uh);
 }
 
+Tensor linalg_eig_backward(const std::vector<torch::autograd::Variable> &grads,
```

Comment:
@Lezcano, have you checked the code of `eig_backward` function? It has lots of useful comments and the math is almost the same, the real input case is treated differently.

I think it would be great to merge the two functions if possible, or at least define a common helper function to use.

Reply:
I looked into that, but since we are going to deprecate `torch.eig` in favour of this one, I think it is not worth the effort.
Mixing the two would make some bits of the code slightly less efficient also.

Reply:
We need to make sure that at least the valuable comments written by @nikitaved are not lost in the future.